### PR TITLE
*: prune deps

### PR DESCRIPTION
**Pull request by nrc**
###  What have you changed?

Removed as many duplicate deps as currently possible (there is still some duplication, but none that are possible to resolve without multiple upstream changes).

This PR takes us from 367 to 327 deps (an 11% improvement). We currently have 5 known security vulnerabilities in our deps (according to `cargo audit`), this PR resolves all of them. This PR also improves compile times: a release build on my MacBook goes from 11:23 to 10:43 (an improvement of 6%). And of course gets us more recent versions of many dependencies.

###  What is the type of the changes?

Pick one of the following and delete the others:

- Engineering (engineering change which doesn't change any feature or fix any issue)

###  How is the PR tested?

Existing tests. Measured compile time.

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No

###  Does this PR affect `tidb-ansible`?

No

PTAL @zhangjinpeng1987 @overvenus

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -215,7 +215,6 @@ clippy: pre-clippy
 		-A clippy::excessive_precision -A clippy::collapsible_if -A clippy::blacklisted_name \
 		-A clippy::needless_range_loop -A clippy::redundant_closure \
 		-A clippy::match_wild_err_arm -A clippy::blacklisted_name -A clippy::redundant_closure_call
-
```
Comment (Hoverbear) [inline Makefile]: Unrelated change?

**Comment by siddontang** [comment]
Great

CI failed
